### PR TITLE
fix: map fullnames to addresses correctly

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
We fetched full-names but they were not mapped back to the addresses.

- [x] map full names back to address
- [x] only return full names for public visibility

About # (link your issue here)
#140 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/app/src/hooks/useFetchFullName.ts [hunk 1]
```diff
@@ -4,18 +4,32 @@ import { gql } from '@apollo/client';
 import { useMongoDbQuery } from './apollo/useMongoDbQuery';
 
 interface UserProfile {
-  fullName?: { display?: string };
+  fullName?: { display?: string; privacy: string };
+  index: {
+    walletAddress: {
+      hash: string;
+      display?: string;
+      value?: string;
```
Comment (sirpy) [inline packages/app/src/hooks/useFetchFullName.ts:12]: value not required

[the remark this answers is not in the document]
Comment (sirpy) [reply packages/app/src/hooks/useFetchFullName.ts:12]: also seems like we are not using UserProfile interface anywhere. so either remove it or use it

**Review by sirpy — CHANGES_REQUESTED** [review]

File: packages/app/src/hooks/useFetchFullName.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/hooks/useFetchFullName.ts:12]: UserProfile interface is used in the UserProfilesResponse interface

File: packages/app/src/hooks/useFetchFullName.ts [hunk 2]
```diff
@@ -27,22 +41,35 @@ export function useFetchFullName(address?: string): string | undefined {
...
+    return acc;
+  }, {});
+
+  const hashedAddresses = Object.keys(addressToHashMapping);
 
   const { data, error } = useMongoDbQuery<UserProfilesResponse>(findProfiles, {
-    variables: { query: { index: { walletAddress: { hash_in: hashedAddresses } } } },
+    variables: {
+      query: {
+        fullName: { privacy: 'public' },
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/hooks/useFetchFullName.ts]: changed

File: packages/app/src/hooks/useFetchFullName.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/hooks/useFetchFullName.ts]: value removed

**Review by sirpy — APPROVED** [review]

**Pull request merged**